File: app/src/main/java/com/example/mandatoryassignment/RegisterActivity.java
package com.example.mandatoryassignment;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.EditText;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.textfield.TextInputEditText;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.auth.UserProfileChangeRequest;

public class RegisterActivity extends AppCompatActivity {

    private TextInputEditText user;
    private TextInputEditText password;
    private TextInputEditText repeatpassword;
    private TextView errorMessage;
    private FirebaseAuth mAuth;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_regiser);
        user=findViewById(R.id.registerUsername);
        password=findViewById(R.id.registerPassword);
        repeatpassword=findViewById(R.id.repeatPassword);
        errorMessage=findViewById(R.id.errorText);
        mAuth=FirebaseAuth.getInstance();
    }

    public void Register(View v)
    {
        if(password.getText().toString().equals(repeatpassword.getText().toString())) {
            mAuth.createUserWithEmailAndPassword(user.getText().toString(), password.getText().toString())
                    .addOnSuccessListener(this, authResult -> {
                        Log.d("banana", "createUserWIthEmail : success");
                        Toast.makeText(RegisterActivity.this, "Successfully Created User.", Toast.LENGTH_SHORT).show();
                        FirebaseUser u = authResult.getUser();
                        goToLogInPage();

                    }).addOnFailureListener(this, e -> {
                Log.w("banana", "createUserWithEmail : failure", e);
                Toast.makeText(RegisterActivity.this, "Registration Failed.", Toast.LENGTH_SHORT).show();
                errorMessage.setText("Registration error: \n" + e.getMessage());
                errorMessage.setVisibility(View.VISIBLE);
            });
        }
        else
        {
            errorMessage.setText("Passwords do not mach");
            errorMessage.setVisibility(View.VISIBLE);
        }
    }
    public void goToLogInPage()
    {
        Intent gotoLogIn= new Intent(this, MainActivity.class);
        startActivity(gotoLogIn);
    }
}

File: app/src/main/java/com/example/mandatoryassignment/RegisterActivity.kt

package com.example.mandatoryassignment

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputEditText
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
//import kotlinx.android.syntetic.main.activity_register.*


class RegisterActivity : AppCompatActivity() {
    private var user: TextInputEditText? = null
    private var password: TextInputEditText? = null
    private var repeatpassword: TextInputEditText? = null
    private var errorMessage: TextView? = null
    private var mAuth: FirebaseAuth? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_regiser)
        user = findViewById(R.id.registerUsername)
        password = findViewById(R.id.registerPassword)
        repeatpassword = findViewById(R.id.repeatPassword)
        errorMessage = findViewById(R.id.errorText)
        mAuth = FirebaseAuth.getInstance()
    }

    @SuppressLint("SetTextI18n")
    fun Register(v: View?) {
        if (password!!.text.toString() == repeatpassword!!.text.toString()) {
            mAuth!!.createUserWithEmailAndPassword(user!!.text.toString(), password!!.text.toString())
                    .addOnSuccessListener(this) { authResult: AuthResult ->
                        Log.d("banana", "createUserWIthEmail : success")
                        Toast.makeText(this@RegisterActivity, "Successfully Created User.", Toast.LENGTH_SHORT).show()
                        val u = authResult.user
                        goToLogInPage()
                    }.addOnFailureListener(this) { e: Exception ->
                        Log.w("banana", "createUserWithEmail : failure", e)
                        Toast.makeText(this@RegisterActivity, "Registration Failed.", Toast.LENGTH_SHORT).show()
                        errorMessage!!.text = """Registration error: ${e.message}""".trimIndent()
                        errorMessage!!.visibility = View.VISIBLE
                    }
        } else {
            errorMessage!!.text = "Passwords do not mach"
            errorMessage!!.visibility = View.VISIBLE
        }
    }

    fun goToLogInPage() {
        val gotoLogIn = Intent(this, MainActivity::class.java)
        startActivity(gotoLogIn)
    }
}